src/mpid/ch4/include/mpidpre.h

```diff
@@ -270,6 +270,8 @@ typedef struct MPIDI_CH4U_win_sync {
     int access_epoch_type;      /* NONE, FENCE, LOCKALL, START,
                                  * LOCK (refer to target_sync). */
     int exposure_epoch_type;    /* NONE, FENCE, POST. */
+    uint32_t assert_mode;       /* bit-vector OR of zero or more of the following integer constant:
```

Comment:
We need to store `NOCHECK` also per target . Every single lock/unlock checks the per-target one, and the other sync calls check the per-window one. Otherwise the following case will not work:
```
lock(assert = 0, P1)
lock(assert = nocheck, P0)
...
unlock(P1)
unlock(P0)
```

Reply:
add target_assert_mode for every target.